pyriemann_qiskit/classification.py

```diff
@@ -0,0 +1,360 @@
+        Contains vectorized test set for target and non-target classes
+
+    Notes
+    -----
+    .. versionadded:: 0.0.1
+
+    Attributes
+    ----------
+    classes_ : list
+        list of classes.
+
+    See Also
+    --------
+    QuanticSVM
+    QuanticVQC
+
+    """
+
+    def __init__(self, target, q_account_token=None, quantum=True,
+                 process_vector=lambda v: v, verbose=True, test_input={}):
+        self.verbose = verbose
+        self._log("Initializing Quantum Classifier")
+        self.test_input = test_input
+        self.process_vector = process_vector
+        self.q_account_token = q_account_token
+        self.target = target
+        self.quantum = quantum
+        # protected field for child classes
+        self._training_input = {}
+
+    def _init_quantum(self):
+        if self.quantum:
+            aqua_globals.random_seed = datetime.now().microsecond
+            self._log("seed = ", aqua_globals.random_seed)
+            if self.q_account_token:
+                self._log("Real quantum computation will be performed")
+                IBMQ.delete_account()
+                IBMQ.save_account(self.q_account_token)
+                IBMQ.load_account()
+                self._log("Getting provider...")
+                self._provider = IBMQ.get_provider(hub='ibm-q')
+            else:
+                self._log("Quantum simulation will be performed")
+                self._backend = BasicAer.get_backend('qasm_simulator')
+        else:
+            self._log("Classical SVM will be performed")
+
+    def _log(self, *values):
+        if self.verbose:
+            print("[QClass] ", *values)
+
+    def _vectorize(self, X):
+        vector = X.reshape(len(X), self._feature_dim)
+        return [self.process_vector(x) for x in vector]
+
+    def _split_target_and_non_target(self, X, y):
+        self._log("""[Warning] Spitting target from non target.
+                 Only binary classification is supported.""")
+        n_matrices = len(X[0])
+        try:
+            n_channels = len(X[0][0])
+        except Exception:
+            n_channels = 1
+        self._feature_dim = n_channels * n_matrices
+        self._log("Feature dimension = ", self._feature_dim)
+        Xta = X[y == self.target]
+        Xnt = X[np.logical_not(y == self.target)]
+        vect_Xta = self._vectorize(Xta)
+        vect_Xnt = self._vectorize(Xnt)
+        self._new_feature_dim = len(vect_Xta[0])
+        self._log("Feature dimension after vector processing = ",
+                  self._new_feature_dim)
+        return (vect_Xta, vect_Xnt)
+
+    def _additional_setup(self):
+        self._log("There is no additional setup.")
+
+    def fit(self, X, y):
+        """Prepare the training data and the quantum backend
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_trials, n_channels, n_channels) |
+                     shape (n_trials, n_channels * n_channels)
+            ndarray of SPD matrices.
```

Comment:
Doc should be completed: it seems that matrices can be provided in raw format (3D array), or after vectorization (2D).
Is it a basic vectorization? or, is it a vectorization after projection in the tangent space?